test/factory/smartWalletFactory.test.ts

```diff
@@ -1,497 +1,557 @@
+          },
+          {
+            callForwarder: smartWalletFactory.address,
+          }
+        );
+
+        const typedDeployData = new TypedDeployRequestData(
+          HARDHAT_CHAIN_ID,
+          smartWalletFactory.address,
+          deployRequest
+        );
+
+        const suffixData = getSuffixData(typedDeployData);
+
+        const privateKey = Buffer.from(
+          owner.privateKey.substring(2, 66),
+          'hex'
+        );
+        const signature = getLocalEip712DeploySignature(
+          typedDeployData,
+          privateKey
+        );
+
+        await smartWalletFactory
+          .connect(worker)
+          .relayedUserSmartWalletCreation(
+            deployRequest.request,
+            suffixData,
+            worker.address,
+            signature
+          );
+
+        const smartWallet = await ethers.getContractAt(
+          'SmartWallet',
+          smartWalletAddress
+        );
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(0);
+        await expect(smartWallet.isInitialized()).to.eventually.be.true;
+      });
+
+      it('should initialize the smart wallet in the expected address paying fee', async function () {
+        const initialWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(initialWorkerBalance).to.be.equal(0);
+        await expect(
+          provider.getCode(smartWalletAddress)
+        ).to.eventually.be.equal('0x');
+
+        await token.mint(1000, smartWalletAddress);
+
+        const amountToBePay = 500;
+
+        const deployRequest = createDeployRequest(
+          {
+            from: owner.address,
+            tokenContract: token.address,
+            tokenAmount: amountToBePay,
+            tokenGas: 55000,
+            recoverer: recoverer,
+            index: index,
+            relayHub: worker.address,
+          },
+          {
+            callForwarder: smartWalletFactory.address,
+          }
+        );
+
+        const typedDeployData = new TypedDeployRequestData(
+          HARDHAT_CHAIN_ID,
+          smartWalletFactory.address,
+          deployRequest
+        );
+
+        const suffixData = getSuffixData(typedDeployData);
+
+        const privateKey = Buffer.from(
+          owner.privateKey.substring(2, 66),
+          'hex'
+        );
+        const signature = getLocalEip712DeploySignature(
+          typedDeployData,
+          privateKey
+        );
+
+        await smartWalletFactory
+          .connect(worker)
+          .relayedUserSmartWalletCreation(
+            deployRequest.request,
+            suffixData,
+            worker.address,
+            signature
+          );
+
+        const smartWallet = await ethers.getContractAt(
+          'SmartWallet',
+          smartWalletAddress
+        );
+
+        const finalWorkerBalance = await token.balanceOf(worker.address);
+
+        expect(finalWorkerBalance).to.be.equal(amountToBePay);
```

Comment:
Instead of testing that the initial worker balance is 0, I'd check that the final balance is equal to the initial balance plus the amount paid

Reply:
updated